html/semantics/embedded-content/the-img-element/relevant-mutations-lazy.html

```diff
@@ -0,0 +1,78 @@
+<!doctype html>
+<title>img relevant mutations, lazy-loaded</title>
+<meta name="timeout" content="long">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="support/relevant-mutations.js"></script>
+<div id=log></div>
+
+<img src="/images/green-2x2.png"> <!-- block the window load event -->
+
+<!-- should invoke update the image data, but omit events for layout changes -->
+<!-- but also see https://github.com/whatwg/html/issues/8492 -->
+
+<img srcset="/images/green-2x2.png 2w" sizes="auto" width="2" loading="lazy" data-desc="width attribute changes">
+
+<img srcset="/images/green-2x2.png 2w" sizes="auto" style="width: 2px" loading="lazy" data-desc="width property changes">
+
+<div style="width: 2px">
+  <img srcset="/images/green-2x2.png 2w" sizes="auto" style="width: 100%" loading="lazy" data-desc="percentage width, CB width changes">
+</div>
+
+<img srcset="/images/green-2x2.png 2w" sizes="auto" style="height: 2px; aspect-ratio: 2 / 2" loading="lazy" data-desc="height property changes (with aspect-ratio)">
+
+<img srcset="/images/green-2x2.png 2w" sizes="auto" style="width: 2px" loading="lazy" data-desc="loading attribute state changes">
+
+<img srcset="/images/green-2x2.png 2w" sizes="auto" style="width: 2px" loading="lazy" data-desc="being rendered changes">
+
+<!-- should not invoke update the image data -->
+
+<img srcset="/images/green-2x2.png 2w" sizes="auto" style="width: 2px" loading="lazy" data-desc="display property changes to inline-block">
+
+<img srcset="/images/green-2x2.png 2w" sizes="auto" style="width: 2px" loading="lazy" data-desc="loading attribute changes to LAZY">
+
+<script>
+const rAF = () => new Promise(resolve => requestAnimationFrame(resolve));
+
+onload = async function() {
+
+  await rAF();
+  await rAF();
+
+  // lazy-loaded images should have fired their first 'load' event at this point.
+
+  t('width attribute changes', function(img) {
+    img.width = '4';
+  }, 'load');
+
+  t('width property changes', function(img) {
+    img.style.width = '4px';
+  }, 'timeout');
+
+  t('percentage width, CB width changes', function(img) {
+    img.parentNode.style.width = '4px';
+  }, 'timeout');
+
+  t('height property changes (with aspect-ratio)', function(img) {
+    img.style.height = '4px';
+  }, 'timeout');
+
+  t('loading attribute state changes', function(img) {
+    img.loading = 'eager';
+  }, 'load');
```

Comment:
If the image is already loaded, why would changing the `loading` attribute trigger a load event?

Reply:
State changes to the `loading` attribute is a relevant mutation per https://github.com/whatwg/html/pull/8008/files#diff-41cf6794ba4200b839c53531555f0f3998df4cbb01a4d5cb0b94e3ca5e23947dR30115-R30116

The "auto" keyword is calculated differently depending on the state of `loading`. But I'm now uncertain if it should be a relevant mutation. https://html.spec.whatwg.org/#the-img-element:lazy-loading-attribute already handles the state changing to Eager, so also making it a relevant mutation is questionable...

@yoavweiss @domfarolino any opinions?

Reply:
Yeah, it seems that that specific algorithm replaces the relevant mutation logic (and is more specific). So I'm favorable of removing it from the relevant mutations.

Reply:
Thanks. Thinking about it again, it's invalid to use `auto` for eager images, so if you want to switch to eager you should also change `sizes` (which is a relevant mutation). OK I'll fix the spec and tests.

Reply:
What would happen if authors don't change `sizes` in that case? Would that modify the image's intrinsic dimensions?

Reply:
Hmm, that runs a risk of being confusing.
I wonder if there's room to define `auto` + `eager` as taking the known dimensions in case they are indeed known (and only fallback to 100vw when they aren't)

Reply:
https://github.com/whatwg/html/pull/8008#discussion_r908614408

I prefer the invalid mutation case being somewhat confusing to introducing non-deterministic loading of eager images.

Reply:
Fair! (and apologies for rehashing a past discussion :D)

Reply:
np :)